packages/ui-components/src/components/Switch/Switch.component.tsx

```diff
@@ -0,0 +1,310 @@
+
+const switchBaseStyles = `
+    jn-rounded-full
+    jn-relative
+    jn-p-0
+    jn-leading-0
+    jn-border
+    jn-g-theme-default
+    focus:jn-outline-none
+    focus:jn-ring-2
+    focus:jn-ring-theme-focus
+    disabled:jn-opacity-50
+    disabled:jn-cursor-not-allowed
+`
+
+const borderBaseStyles = `
+    jn-border-theme-switch-default
+`
+
+const switchWrapperStyles = `
+    jn-flex
+    jn-flex-row
+    jn-items-center
+`
+
+const handleBaseStyles = `
+    jn-inline-block
+    jn-absolute
+    jn-top-[1px]
+    jn-rounded-full
+    jn-bg-theme-switch-handle
+    jn-border-theme-default
+`
+
+const handleOnStyles = `
+    jn-right-[1px] 
+    jn-bg-theme-switch-handle-checked
+`
+
+const handleOffStyles = `
+    jn-left-[1px]
+`
+
+const validBaseStyles = `
+    jn-border-theme-success
+`
+
+const invalidBaseStyles = `
+    jn-border-theme-error
+`
+
+const iconBaseStyles = `
+    jn-inline-block 
+    jn-ml-1 
+    jn-leading-1
+    jn-mt-[-.2rem]
+`
+
+const hintBaseStyles = `
+    jn-mt-0
+`
+
+type SwitchSize = "small" | "default" | "large"
+
+export interface SwitchProps extends Omit<React.ComponentPropsWithoutRef<"button">, "ref"> {
+  /**
+   * HTML name attribute for the switch button
+   */
+  name?: string
+  /**
+   * HTML id attribute for the switch button
+   */
+  id?: string
+  /**
+   * Label to display next to the Switch component
+   */
+  label?: string
+  /**
+   * Indicates if the Switch is required
+   */
+  required?: boolean
+  /**
+   * Specifies the size of the Switch component
+   */
+  size?: SwitchSize
+  /**
+   * Sets the initial checked state of the Switch component for use in uncontrolled mode
+   */
+  on?: boolean
+  /**
+   * Disabled state of the Switch component
+   */
+  disabled?: boolean
+  /**
+   * Indicates if the Switch has an error state
+   */
+  invalid?: boolean
+  /**
+   * Indicates if the Switch has been validated successfully
+   */
+  valid?: boolean
+  /**
+   * Help text to provide additional information about the Switch
+   */
+  helptext?: React.ReactNode
+  /**
+   * Text to display when the Switch has validation errors
+   */
+  errortext?: React.ReactNode
+  /**
+   * Text to display when the Switch has been validated successfully
+   */
+  successtext?: React.ReactNode
+  /**
+   * Custom class name to apply to the internal button element
+   */
+  className?: string
+  /**
+   * Callback function to handle changes to the switch state
+   */
+  onChange?: (_event: React.ChangeEvent<HTMLButtonElement>) => void
+  /**
+   * Callback function to handle click events on the switch
+   */
+  onClick?: (_event: React.MouseEvent<HTMLButtonElement>) => void
+  /**
+   * Custom class name to apply to the wrapper element
+   * Useful for styling and positioning the outermost element of the component
+   */
+  wrapperClassName?: string
+}
+
+const isNotEmptyString = (str: unknown): boolean => !(typeof str === "string" && str.trim().length === 0)
```

Comment:
why string here is unknown?

Reply:
Ooops, overlooked as was a little more complicated to figure out. Thank You!